### ISSUE TITLE
App Root Page: when I click the CTA buttons I am directed to a new organization request form

### ISSUE THREAD

**Issue by kasugaijin** [rubyforgood/homeward-tails#1009]
The CTA buttons on the app root page should take the user to a form for organizations to fill out so they can request access to the app (see how we implemented the Feedback form available in the footer and on dashboards). When the user submits this form, it should send an email to the dev email account (TDB) with a title **New Organization Request** and provide the details from the form, below. 

This email will inform the devs that they need to vet the organization, discuss their preferred slug, and then run the service to create a new organization.

The form should collect this information:
- Organization Name
- Your name (the person filling out the form) (we don't need to separate first and last, it's just for conversational purposes)
- Organization Phone Number
- Organization email
- Country
- City/Town
- Province State

Please also change the CTA button text to **Request an account**
Please also update the Org footer link to point to this new form
![Image](https://github.com/user-attachments/assets/d8ecdd00-269f-4de3-81eb-705243914e0d)

App root page CTA buttons x 3
![Image](https://github.com/user-attachments/assets/9ff08720-ac9e-40a4-8d53-bdc88433e5d2)
![Image](https://github.com/user-attachments/assets/1cdb363a-10e0-4990-abdc-6f25cb0c9fd7)

**Comment by DreamVenerable** [comment]
I'm interested ✋🏻

**Comment by kasugaijin** [comment]
@DreamVenerable all yours

**Comment by kasugaijin** [comment]
@DreamVenerable how's this going? Let me know if you have questions.

**Comment by DreamVenerable** [comment]
@kasugaijin Honestly I've been a bit busy, hence the lack of progress. 

If I find myself stuck or unable to complete it I'll inform you.

**Comment by DreamVenerable** [comment]
@kasugaijin Should I create a new model or handle it from the controller?

**Comment by kasugaijin** [comment]
@DreamVenerable please add a new controller - see `app/controllers/contacts_controller.rb` for an example.  We don't need a model as the data will be sent to email. 

I think we can call it something like `app/controllers/organization_acount_requests_controller.rb` unless you can think of a better name. Please also add the route `organization_account_request`, and view for this.

I also note that this will be the third contact form we are adding to the `app/controller` scope. I think there is a good argument to add a module like `contact_forms` in the controller and views so we can keep these controllers and views together. We can move all three controllers/views to this module. Let me know if you want to tackle that also...or we can put in a new issue. 

```
scope module: 'admin' do
  resources :articles, :comments
end
```

**Comment by DreamVenerable** [comment]
I made a new controller called OrganizationRequestsController, but I like your naming better as it's more clear.

Regarding the refactoring, I'll see to it after finishing this issue.

**Comment by kasugaijin** [comment]
@DreamVenerable sounds good!
FYI @Aaryanpal as you are working on one of the contacts controllers

**Comment by DreamVenerable** [comment]
@kasugaijin Hey, sorry I've been busy lately. Can you keep it open so that if someone wants to work on it can get assigned?

**Comment by kasugaijin** [comment]
@DreamVenerable For sure. Issue is ready for someone to be assigned.

**Comment by Meri-MG** [comment]
hi, @kasugaijin maybe i could start working on this one once the other pr is merged?

**Comment by kasugaijin** [comment]
@Meri-MG for sure!

**Comment by kasugaijin** [comment]
@Meri-MG let me know if you have any questions :)

**Comment by kasugaijin** [comment]
Ah right. There is this middleware in lib/middleware/organization_middleware.rb that sets the tenant on each request. So I think, providing there is no org slug in the URL e.g., `/baja` there should not be an org scope to worry about. We shouldn't need an org scope for this route either. Ideally this is handled by a single `OrganizationRequestsController` scoped outside of the organization context.

**Comment by Meri-MG** [comment]
Hello, @kasugaijin, I think I misunderstood something here... so on my current branch i was creating an organization, was getting slug null error so i added one more field where slug is populated automatically by stimulus, but then, today i read the description of the issue again and figured i didn't have to do any of these...  For now, I'm wondering if i create a new model instead of creating an Organization and then the dev team creates an organization based on the provided info? 

[here is my branch with the current changes](https://github.com/Meri-MG/pet-rescue/tree/1009-direct-cta-buttons-to-form)

**Comment by kasugaijin** [comment]
@Meri-MG yes all this needs to do is show a form that when submitted sends an email to the devs. The organization is created later by the devs!

**Comment by Meri-MG** [comment]
Hello, @kasugaijin! I'm having trouble passing the nested location attributes from the controller to the mailer. I'm planning to open a draft PR, and if you have time, could you have a look? Another issue I'm having is with the system test. Choosing a province managed by the stimulus controller isn't working in the test.

**Comment by kasugaijin** [comment]
@Meri-MG for sure put up a draft and I can take a look. I’m also happy with an integration test over a system test. We should be able to pass Parma easily in a request.
